config/deals.go

```diff
@@ -0,0 +1,7 @@
+package config
+
+type DealsConfig struct {
```

Comment:
`Deal`, also, let's drop the `Config` suffix on all the config struct

Reply:
For now I've just renamed to DealConfig

Reply:
Well, I use the package name and read with the struct `Config.Deal`. Also, for the case of instance name, I support the suffix.

```
type Estuary struct  { 
  DealConfig Config.Deal 
}

Reply:
Umm... OK. Since Deal and Estuary are in the same package, my compiler complains about config.Deal. I'm probably missing something. Almost nobody will ever use the struct name (because most accesses of the struct members will be via the enclosing struct). I think it is wrong to choose the shorter name for the struct name, therefore. But, I really really need to get this merged so I'll go ahead and rename all the members to 'XConfig' and all the structs to 'X'.

Reply:
No, that is an example of how it will be used outside of a config package

Reply:
```
type Estuary struct  { 
  DealConfig Config.Deal
}

the above assumes  Estuary is in another package, that is using the deal config - in this case the suffix works. 

```
type Estuary struct  { 
  Deal Deal 
}

the above assumes  Estuary and Deal are both in the config package, in this case, we do not need the suffix for both - when external packages uses them, the external package will have a instance name with the suffix like;

```
type External struct  { 
  EstuaryConfig Config.Estuary
  or 
  Config Config.Estuary
}

Reply:
if this is too much trouble, we can ignore it and get this merged. We can revisit it later.

Reply:
Lastly, do you have a dump of a sample `json` file of these configs? I am curios how it looks like when marshalled to a file. Are you going to add json tags to the structs?

Reply:
ok